ast-nodes/range.dart

```diff
@@ -9,7 +11,30 @@ class Range implements Node {
 
   Range(this.start, this.end);
 
-  // TODO: implement .parse()
+  factory Range.parse(Iterable<Token> tokens) {
+
+    var iterator = tokens.iterator;
+    if(!iterator.moveNext()){
+      throw SyntaxError(iterator.current, "Expected '..' in range");
+    }
+
+    var expStart = Expression.parse(
+        consumeUntil(iterator, RegExp("\^"+RegExp.escape("..")+"\$"))
```

Comment:
`var expStart` will include the keyword `in` in such a case. At the moment of creation iterator points at 0th element of the Iterable object, so after you moved it on line 17, it started pointing at `in`. `consumeUntil` function includes the element that iterator is pointing at the time of function call.

Reply:
And you forgot `reverse` keyword btw.

Reply:
From our project description range doesn't include in and reverse keywords)